test/blackbox-tests/test-cases/trace-file.t/run.t

```diff
@@ -3,6 +3,7 @@
 This captures the commands that are being run:
 
   $ <trace.json grep '"X"' | cut -c 2- | sed -E 's/:[0-9]+/:.../g'
+  {"args":{"process_args":["printconf","conf"],"pid":...},"ph":"X","dur":...,"name":"ocamlfind","cat":"process","ts":...,"pid":...,"tid":...}
```

Comment:
Are we always running `ocamlfind printconf` now?

Reply:
I think we're always running `ocamlfind printconf conf` now if `OCAMLFIND_CONF` is not set. This is akin to the approach in https://github.com/ocaml/dune/pull/1707

For OPAM environments, I think we know that we can find a config, if it exists, in `$OPAM_SWITCH_PREFIX/lib/findlib.conf`. If we're to prefer `findlib.conf` over `OCAMLPATH` then I believe we need to call this command.

As per the findlib reference manual, `OCAMLPATH` is meant to be an additional, prepended list of paths:

> OCAMLPATH
This variable may contain an additional search path for package directories. It is treated as if the directories were prepended to the configuration variable path.

Reply:
> I think we're always running ocamlfind printconf conf now if OCAMLFIND_CONF is not set. This is akin to the approach in https://github.com/ocaml/dune/pull/1707

Always? Do you mean only for cross compilation? I'm pretty sure we don't call `ocamlfind printconf` in normal builds. 1707 indeed introduces the same issue as this PR

Reply:
Sorry, I mean "now" as in this PR. 

I thought we wanted to read the default findlib.conf file in all cases, not just when cross-compiling.

Reply:
It would be good to know where it is, but running ocamlfind by default to get it doesn't seem like the right way to go. Running ocamlfind by default is something that was explicitly removed in 2.0, so I'd prefer not to bring it back.

See the comment here https://github.com/ocaml/dune/pull/1707#issuecomment-463147994 and the discussion leading to it

Reply:
I see. That comment is super helpful. I'll make these changes:

> [...] go for full findlib support whenever OCAMLFIND_CONF or OCAMLFIND_TOOLCHAIN are set.

Reply:
Alright, in my most recent push I reverted to only calling `ocamlfind printconf conf` when looking for the configuration for the target context.